src/l3.c

```diff
@@ -36,6 +44,14 @@
 #define L3_STATIC_ASSERT    _Static_assert
 #endif
 
+#if __APPLE__
+#define L3_THREAD_LOCAL
```

Comment:
I don't think this will work. It'll mean the cached thread id is not thread local, so every thread will report the thread-id of the first thread to call one of our logging functions.

Given that the main idea behind this was a logging library suitable for tracking down race conditions, this is bad.

I'm surprised it's necessary - it's standard C11 so I'd have thought it would be supported on Mac. https://en.cppreference.com/w/c/thread/thread_local

Reply:
Yes, you are right. 

Early on when I started this port work, I was a bit lazy trying to figure out the OSX-specific variation to get thread-ID, so just slammed-in something to get things compiling.

I have figured out what to do here, and have also enhanced the tests to verify that the generate thread-ID is != 0.

```
333 def verify_output_lists(nentries:int,
334                         exp_nentries:int,
335                         tid_list:list,
[...]
356     # As in the unit-test a single thread does slow-logging all log-entries
357     # must have the same decoded TID value. And, make sure that L3-logging
358     # code has generated a valid thread-ID, which should be non-zero.
359     assert tid_list[0] != 0
360     for tid in tid_list:
361         assert tid == tid_list[0]
```

Have cross-checked that the tests fail without the change to correct thread-ID.